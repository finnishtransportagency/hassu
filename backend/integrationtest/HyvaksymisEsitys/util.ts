import { config } from "../../src/config";
import { getS3Client } from "../../src/aws/client";
import { DeleteObjectsCommand, ListObjectsV2Command, ListObjectsV2CommandOutput, PutObjectCommand } from "@aws-sdk/client-s3";
import { createPresignedPost } from "@aws-sdk/s3-presigned-post";
import { adaptFileName } from "../../src/tiedostot/paths";
import { getAxios } from "../../src/aws/monitoring";
import FormData from "form-data";

export async function insertYllapitoFileToS3(pathInS3: string) {
  if (pathInS3.includes(config.yllapitoBucketName)) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  const buffer = Buffer.from("Test");
  await getS3Client().send(
    new PutObjectCommand({
      Body: buffer,
      Bucket: config.yllapitoBucketName,
      Key: pathInS3,
    })
  );
}

export async function insertUploadFileToS3(uuid: string, filename: string) {
  const fileNameWithPath = `${uuid}/${adaptFileName(filename)}`;
  const presignedPost = await createPresignedPost(getS3Client(), {
    Bucket: config.uploadBucketName,
    Key: fileNameWithPath,
    Expires: 600,
  });
  const fields = presignedPost.fields;
  const form = new FormData();
  Object.keys(fields).forEach((key) => {
    form.append(key, fields[key]);
  });
  form.append("file", "Test");
  const axios = getAxios();
  const lengthSync = form.getLengthSync();

  await axios.post(presignedPost.url, form, {
    headers: {
      "Content-Length": lengthSync,
      ...form.getHeaders(),
    },
  });
}

export async function deleteYllapitoFiles(sourcePrefix?: string) {
  if (sourcePrefix && sourcePrefix.includes(config.yllapitoBucketName)) {
    throw new Error("sourcePrefix ei saa sisältää yllapitoBucketNamea");
  }
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: sourcePrefix,
        ContinuationToken,
      })
    );
    await deleteFiles(Contents.map(({ Key }) => Key as string));

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

export async function emptyUploadFiles() {
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.uploadBucketName,
        ContinuationToken,
      })
    );
    await deleteFiles(
      Contents.map(({ Key }) => Key as string),
      config.uploadBucketName
    );

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

async function deleteFiles(paths: string[], bucket: string = config.yllapitoBucketName) {
  if (paths.some((path) => path.includes(config.yllapitoBucketName))) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  await Promise.all(
    // DeleteObjectsCommand takes at most 1000 items
    getChunksOfThousand(paths).map((paths) =>
      getS3Client().send(
        new DeleteObjectsCommand({
          Bucket: bucket,
          Delete: { Objects: paths.map((path) => ({ Key: path })) },
        })
      )
    )
  );
}

function getChunksOfThousand<T>(array: T[]): T[][] {
  const chunks: T[][] = [];
  for (let i = 0; i < array.length; i += 1000) {
    chunks.push(array.slice(i, i + 1000));
  }
  return chunks;
}

export async function getYllapitoFilesUnderPath(pathInS3: string): Promise<string[]> {
  const s3 = getS3Client();
  const files: string[] = [];
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: pathInS3,
        ContinuationToken,
      })
    );
    files.push(...Contents.map(({ Key }) => Key as string));
    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
  return files;
}
