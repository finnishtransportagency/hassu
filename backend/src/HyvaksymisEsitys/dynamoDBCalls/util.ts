import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
import { log } from "../../logger";
import { getDynamoDBDocumentClient } from "../../aws/client";
import { ConditionalCheckFailedException } from "@aws-sdk/client-dynamodb";
import { SimultaneousUpdateError } from "hassu-common/error";

export async function sendUpdateCommandToDynamoDB(params: UpdateCommand): Promise<void> {
  if (log.isLevelEnabled("debug")) {
    log.debug("Updating projekti to Hassu with params", { params });
  }
  try {
    const dynamoDBDocumentClient = getDynamoDBDocumentClient();
    await dynamoDBDocumentClient.send(params);
  } catch (e) {
    if (e instanceof ConditionalCheckFailedException) {
      throw new SimultaneousUpdateError("Projektia on p√§ivitetty tietokannassa. Lataa projekti uudelleen.");
    }
    log.error(e instanceof Error ? e.message : String(e), { params });
    throw e;
  }
}
