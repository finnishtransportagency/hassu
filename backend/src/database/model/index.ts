export * from "./common";
export * from "./projekti";
export * from "./suunnitteluVaihe";
export * from "./nahtavillaoloVaihe";
export * from "./hyvaksymisPaatosVaihe";
