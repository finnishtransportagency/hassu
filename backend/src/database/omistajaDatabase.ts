import { log } from "../logger";
import { config } from "../config";
import { getDynamoDBDocumentClient } from "../aws/client";
import {
  PutCommand,
  PutCommandOutput,
  ScanCommand,
  ScanCommandOutput,
  QueryCommand,
  BatchWriteCommand,
  UpdateCommand,
  TransactWriteCommand,
  TransactWriteCommandInput,
} from "@aws-sdk/lib-dynamodb";
import { chunkArray } from "./chunkArray";

export type OmistajaKey = {
  oid: string;
  id: string;
};

export type DBOmistaja = {
  id: string;
  oid: string;
  kiinteistotunnus?: string | null;
  kayttooikeusyksikkotunnus?: string | null;
  lisatty: string;
  paivitetty?: string | null;
  etunimet?: string | null;
  sukunimi?: string | null;
  nimi?: string | null;
  henkilotunnus?: string;
  ytunnus?: string;
  jakeluosoite?: string | null;
  postinumero?: string | null;
  paikkakunta?: string | null;
  maakoodi?: string | null;
  expires?: number;
  kaytossa: boolean;
  suomifiLahetys?: boolean;
  lahetykset?: { tila: "OK" | "VIRHE"; lahetysaika: string }[];
  userCreated?: boolean;
};

export type OmistajaScanResult = {
  startKey: OmistajaKey | undefined;
  omistajat: DBOmistaja[];
};

type TransactionItem = Exclude<TransactWriteCommandInput["TransactItems"], undefined>[0];

class OmistajaDatabase {
  private tableName: string;
  constructor(tableName: string) {
    this.tableName = tableName;
  }

  async createOmistaja(omistaja: DBOmistaja): Promise<PutCommandOutput> {
    const params = new PutCommand({
      TableName: this.tableName,
      Item: omistaja,
    });
    return getDynamoDBDocumentClient().send(params);
  }

  async scanOmistajat(startKey?: OmistajaKey): Promise<OmistajaScanResult> {
    try {
      const params = new ScanCommand({
        TableName: this.tableName,
        Limit: 50,
        ExclusiveStartKey: startKey,
      });
      const data: ScanCommandOutput = await getDynamoDBDocumentClient().send(params);
      return {
        omistajat: data.Items as DBOmistaja[],
        startKey: (data.LastEvaluatedKey as OmistajaKey) ?? undefined,
      };
    } catch (e) {
      log.error(e);
      throw e;
    }
  }

  async haeProjektinKaytossaolevatOmistajat(oid: string, expression?: string): Promise<DBOmistaja[]> {
    let lastEvaluatedKey: Record<string, any> | undefined;
    const omistajat: DBOmistaja[] = [];
    let data;
    do {
      const command = new QueryCommand({
        TableName: this.tableName,
        KeyConditionExpression: "#oid = :oid",
        ExpressionAttributeValues: {
          ":oid": oid,
          ":kaytossa": true,
        },
        ExpressionAttributeNames: {
          "#oid": "oid",
          "#kaytossa": "kaytossa",
        },
        FilterExpression: "#kaytossa = :kaytossa",
        ExclusiveStartKey: lastEvaluatedKey,
        ProjectionExpression: expression,
      });
      data = await getDynamoDBDocumentClient().send(command);
      lastEvaluatedKey = data?.LastEvaluatedKey;
      omistajat.push(...((data?.Items as DBOmistaja[]) ?? []));
    } while (lastEvaluatedKey !== undefined);
    return omistajat;
  }

  async poistaOmistajaKaytosta(oid: string, id: string): Promise<void> {
    const params = new UpdateCommand({
      TableName: this.tableName,
      Key: {
        oid,
        id,
      },
      UpdateExpression: "SET #kaytossa = :kaytossa",
      ExpressionAttributeNames: {
        "#kaytossa": "kaytossa",
      },
      ExpressionAttributeValues: {
        ":kaytossa": false,
      },
    });
    await getDynamoDBDocumentClient().send(params);
  }

  async vaihdaProjektinKaytossaolevatOmistajat(oid: string, lisattavatOmistajat: DBOmistaja[]): Promise<void> {
    try {
      const items = await this.haeProjektinKaytossaolevatOmistajat(oid);
      if (items.length) {
        log.info("Otetaan käytöstä " + items.length + " omistaja(a)");
        const transactItems = items.map<TransactionItem>((item) => ({
          Update: {
            ExpressionAttributeNames: {
              "#kaytossa": "kaytossa",
            },
            ExpressionAttributeValues: {
              ":kaytossa": false,
            },
            UpdateExpression: "set #kaytossa = :kaytossa",
            TableName: this.tableName,
            Key: {
              id: item.id,
              oid: item.oid,
            },
          },
        }));

        const oldOmistajatChunks = chunkArray(transactItems, 25);

        for (const chunk of oldOmistajatChunks) {
          const transactCommand = new TransactWriteCommand({
            TransactItems: chunk,
          });
          await getDynamoDBDocumentClient().send(transactCommand);
        }
      }
      log.info("Lisätään " + lisattavatOmistajat.length + " omistaja(a)");
      const newTransactItems = lisattavatOmistajat.map<TransactionItem>((item) => ({
        Put: {
          TableName: this.tableName,
          Item: item,
        },
      }));
      const newOmistajatChunks = chunkArray(newTransactItems, 25);
      for (const chunk of newOmistajatChunks) {
        const transactCommand = new TransactWriteCommand({
          TransactItems: chunk,
        });
        await getDynamoDBDocumentClient().send(transactCommand);
      }
    } catch (error) {
      log.error("Projektin kiinteistönomistajien korvaaminen epäonnistui");
      throw error;
    }
  }

  async deleteOmistajatByOid(oid: string) {
    if (config.env !== "prod") {
      let lastEvaluatedKey: Record<string, any> | undefined;
      const omistajat: Record<string, any>[] = [];
      do {
        const command = new QueryCommand({
          TableName: this.tableName,
          KeyConditionExpression: "#oid = :oid",
          ExpressionAttributeValues: {
            ":oid": oid,
          },
          ExpressionAttributeNames: {
            "#oid": "oid",
          },
          ProjectionExpression: "id",
          ExclusiveStartKey: lastEvaluatedKey,
        });
        const data = await getDynamoDBDocumentClient().send(command);
        lastEvaluatedKey = data.LastEvaluatedKey;
        omistajat.push(...(data.Items ?? []));
      } while (lastEvaluatedKey !== undefined);
      for (const chunk of chunkArray(omistajat, 25)) {
        const deleteRequests = chunk.map((omistaja) => ({
          DeleteRequest: {
            Key: { oid, id: omistaja.id },
          },
        }));
        await getDynamoDBDocumentClient().send(
          new BatchWriteCommand({
            RequestItems: {
              [this.tableName]: deleteRequests,
            },
          })
        );
      }
    }
  }
}

export const omistajaDatabase = new OmistajaDatabase(config.kiinteistonomistajaTableName ?? "missing");
