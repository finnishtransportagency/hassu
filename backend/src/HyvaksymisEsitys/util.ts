import { IllegalArgumentError } from "hassu-common/error";
import { DBProjekti } from "../database/model";
import { projektiDatabase } from "../database/projektiDatabase";
import { requirePermissionLuku } from "../user";
import { NykyinenKayttaja } from "hassu-common/graphql/apiModel";

export async function varmistaLukuoikeusJaHaeProjekti(
  oid: string
): Promise<{ projektiInDB: DBProjekti; nykyinenKayttaja: NykyinenKayttaja }> {
  const nykyinenKayttaja = requirePermissionLuku();
  const projektiInDB = await projektiDatabase.loadProjektiByOid(oid);
  if (!projektiInDB) {
    throw new IllegalArgumentError(`Projektia oid:lla ${oid} ei l√∂ydy`);
  }
  return { projektiInDB, nykyinenKayttaja };
}
