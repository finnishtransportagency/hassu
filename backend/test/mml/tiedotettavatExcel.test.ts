import { Vaihe } from "hassu-common/graphql/apiModel";
import { DBProjekti, DBVaylaUser } from "../../src/database/model";
import { generateExcel, generateExcelByQuery } from "../../src/mml/tiedotettavatExcel";
import { mockClient } from "aws-sdk-client-mock";
import { DynamoDBDocumentClient, GetCommand, QueryCommand } from "@aws-sdk/lib-dynamodb";
import { DBOmistaja } from "../../src/database/omistajaDatabase";
import readXlsxFile from "read-excel-file/node";
import { expect } from "chai";
import { identifyMockUser } from "../../src/user/userService";
import { getKiinteistonomistajaTableName, getMuistuttajaTableName } from "../../src/util/environment";
import { DBMuistuttaja } from "../../src/muistutus/muistutusHandler";
import fs from "fs";

const projekti: Partial<DBProjekti> = {
  oid: "1.2.3",
  kayttoOikeudet: [{ kayttajatunnus: "testuid" } as unknown as DBVaylaUser],
};
const omistaja1: DBOmistaja = {
  id: "1",
  kaytossa: true,
  kiinteistotunnus: "12345600010291",
  lisatty: "2024-02-21 10:01:12+02:00",
  oid: projekti.oid!,
  etunimet: "Etunimi",
  sukunimi: "Sukunimi",
  jakeluosoite: "Osoite 1",
  postinumero: "00100",
  paikkakunta: "Helsinki",
  suomifiLahetys: true,
};
const omistaja2: DBOmistaja = {
  id: "2",
  kaytossa: true,
  kiinteistotunnus: "12345600010291",
  lisatty: "2024-02-21 10:01:12+02:00",
  oid: projekti.oid!,
  etunimet: "Etunimi2",
  sukunimi: "Sukunimi2",
  jakeluosoite: "Osoite 2",
  postinumero: "01600",
  paikkakunta: "Vantaa",
  suomifiLahetys: true,
};
const omistaja3: DBOmistaja = {
  id: "3",
  kaytossa: true,
  kiinteistotunnus: "12345600010291",
  lisatty: "2024-02-21 10:01:12+02:00",
  oid: projekti.oid!,
  etunimet: "Etunimi3",
  sukunimi: "Sukunimi3",
  jakeluosoite: "Osoite 3",
  postinumero: "02600",
  paikkakunta: "Espoo",
  suomifiLahetys: false,
};
const omistaja4: DBOmistaja = {
  id: "4",
  kaytossa: true,
  kiinteistotunnus: "12345600010291",
  lisatty: "2024-02-21 10:01:12+02:00",
  oid: projekti.oid!,
  nimi: "Yritys Oy Ab",
  jakeluosoite: "Osoite 4",
  postinumero: "96101",
  paikkakunta: "Rovaniemi",
  suomifiLahetys: false,
};
const omistaja5: DBOmistaja = {
  id: "5",
  kaytossa: true,
  kiinteistotunnus: "04105600010000",
  lisatty: "2024-02-21 10:01:12+02:00",
  paivitetty: "2024-02-22 08:43:12+02:00",
  oid: projekti.oid!,
  suomifiLahetys: false,
};
const muistuttaja1: DBMuistuttaja = {
  id: "1",
  expires: 0,
  lisatty: "2024-02-21 10:01:12+02:00",
  oid: projekti.oid!,
  etunimi: "Tytti",
  sukunimi: "Testaaja",
  henkilotunnus: "111111-112A",
  lahiosoite: "Osoite 456",
  postinumero: "02600",
  postitoimipaikka: "Espoo",
};
const muistuttaja2: DBMuistuttaja = {
  id: "2",
  expires: 0,
  lisatty: "2024-02-21 10:01:12+02:00",
  oid: projekti.oid!,
  etunimi: "Matti",
  sukunimi: "Testaaja",
  henkilotunnus: "111111-111A",
  lahiosoite: "Osoite 123",
  postinumero: "01600",
  postitoimipaikka: "Vantaa",
};
const muistuttaja3: DBMuistuttaja = {
  id: "3",
  expires: 0,
  lisatty: "2024-02-21 10:01:12+02:00",
  oid: projekti.oid!,
  etunimi: "Teppo",
  sukunimi: "Muistuttaja",
  henkilotunnus: "",
  lahiosoite: "Muistuttajan osoite",
  postinumero: "00100",
  postitoimipaikka: "Helsinki",
};
describe("tiedotettavatExcel", () => {
  before(() => {
    identifyMockUser({ roolit: ["hassu_admin"], etunimi: "Test", sukunimi: "Test", uid: "testuid", __typename: "NykyinenKayttaja" });
    mockClient(DynamoDBDocumentClient)
      .on(QueryCommand, { TableName: getKiinteistonomistajaTableName() })
      .resolves({ Items: [omistaja1, omistaja2, omistaja3, omistaja4, omistaja5] })
      .on(QueryCommand, { TableName: getMuistuttajaTableName() })
      .resolves({ Items: [muistuttaja1, muistuttaja2, muistuttaja3] })
      .on(GetCommand)
      .resolves({ Item: projekti });
  });
  it("tallenna kiinteistön omistajat excel tiedostoon nähtävilläolo", async () => {
    const buffer = await generateExcel(projekti as DBProjekti, true, Vaihe.NAHTAVILLAOLO, "2024-02-21");
    fs.writeFileSync(__dirname + "/maanomistajaluettelo_nahtavillaolo.xlsx", buffer);
    let rows = await readXlsxFile(buffer, { sheet: "Suomi.fi kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
    rows = await readXlsxFile(buffer, { sheet: "Muut kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
  });
  it("tallenna kiinteistön omistajat excel tiedostoon hyväksymispäätös", async () => {
    const buffer = await generateExcel(projekti as DBProjekti, true, Vaihe.HYVAKSYMISPAATOS, "2024-02-21");
    fs.writeFileSync(__dirname + "/maanomistajaluettelo_hyvaksymispaatos.xlsx", buffer);
    let rows = await readXlsxFile(buffer, { sheet: "Suomi.fi kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
    rows = await readXlsxFile(buffer, { sheet: "Muut kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
    rows = await readXlsxFile(buffer, { sheet: "Suomi.fi muistuttajat" });
    expect(rows).toMatchSnapshot();
    rows = await readXlsxFile(buffer, { sheet: "Muut muistuttajat" });
    expect(rows).toMatchSnapshot();
  });
  it("tallenna kiinteistön omistajat excel tiedostoon Suomi.fi", async () => {
    const buffer = await generateExcel(projekti as DBProjekti, true, undefined, undefined, true);
    const rows = await readXlsxFile(buffer, { sheet: "Suomi.fi kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
  });
  it("tallenna kiinteistön omistajat excel tiedostoon muut", async () => {
    const buffer = await generateExcel(projekti as DBProjekti, true, undefined, undefined, false);
    const rows = await readXlsxFile(buffer, { sheet: "Muut kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
  });
  it("tallenna kiinteistön omistajat excel tiedostoon graphql Suomi.fi", async () => {
    const excel = await generateExcelByQuery({ kiinteisto: true, oid: "1.2.3", suomifi: true });
    const rows = await readXlsxFile(Buffer.from(excel.sisalto, "base64"), { sheet: "Suomi.fi kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
  });
  it("tallenna kiinteistön omistajat excel tiedostoon graphql muut", async () => {
    const excel = await generateExcelByQuery({ kiinteisto: true, oid: "1.2.3", suomifi: false });
    const rows = await readXlsxFile(Buffer.from(excel.sisalto, "base64"), { sheet: "Muut kiinteistön omistajat" });
    expect(rows).toMatchSnapshot();
  });
});
