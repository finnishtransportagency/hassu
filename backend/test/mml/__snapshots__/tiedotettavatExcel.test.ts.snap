// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon Suomi.fi 1`] = `
Array [
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi Sukunimi",
    "Osoite 1",
    "00100",
    "Helsinki",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
  Array [
    "123-456-1-291",
    "Etunimi2 Sukunimi2",
    "Osoite 2",
    "01600",
    "Vantaa",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon graphql Suomi.fi 1`] = `
Array [
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi Sukunimi",
    "Osoite 1",
    "00100",
    "Helsinki",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
  Array [
    "123-456-1-291",
    "Etunimi2 Sukunimi2",
    "Osoite 2",
    "01600",
    "Vantaa",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon graphql muut 1`] = `
Array [
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi3 Sukunimi3",
    "Osoite 3",
    "02600",
    "Espoo",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "123-456-1-291",
    "Yritys Oy Ab",
    "Osoite 4",
    "96101",
    "Rovaniemi",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "41-56-1-0",
    null,
    null,
    null,
    null,
    "22.02.2024 08:43:12+02:00",
    "Kirjeitse",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon hyväksymispäätös 1`] = `
Array [
  Array [
    "Kuulutus suunnitelman hyväksymisestä",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "21.02.2024",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistönomistajien tiedotus Suomi.fi -palvelulla",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi Sukunimi",
    "Osoite 1",
    "00100",
    "Helsinki",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
  Array [
    "123-456-1-291",
    "Etunimi2 Sukunimi2",
    "Osoite 2",
    "01600",
    "Vantaa",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon hyväksymispäätös 2`] = `
Array [
  Array [
    "Kuulutus suunnitelman hyväksymisestä",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "21.02.2024",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistönomistajien tiedotus muilla tavoin",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi3 Sukunimi3",
    "Osoite 3",
    "02600",
    "Espoo",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "123-456-1-291",
    "Yritys Oy Ab",
    "Osoite 4",
    "96101",
    "Rovaniemi",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "41-56-1-0",
    null,
    null,
    null,
    null,
    "22.02.2024 08:43:12+02:00",
    "Kirjeitse",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon hyväksymispäätös 3`] = `
Array [
  Array [
    "Kuulutus suunnitelman hyväksymisestä",
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "21.02.2024",
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Muistuttajien tiedotus Suomi.fi -palvelulla",
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Muistuttajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "Tytti Testaaja",
    "Osoite 456",
    "02600",
    "Espoo",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
  Array [
    "Matti Testaaja",
    "Osoite 123",
    "01600",
    "Vantaa",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon hyväksymispäätös 4`] = `
Array [
  Array [
    "Kuulutus suunnitelman hyväksymisestä",
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "21.02.2024",
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Muistuttajien tiedotus muilla tavoin",
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Muistuttajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "Teppo Muistuttaja",
    "Muistuttajan osoite",
    "00100",
    "Helsinki",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon muut 1`] = `
Array [
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi3 Sukunimi3",
    "Osoite 3",
    "02600",
    "Espoo",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "123-456-1-291",
    "Yritys Oy Ab",
    "Osoite 4",
    "96101",
    "Rovaniemi",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "41-56-1-0",
    null,
    null,
    null,
    null,
    "22.02.2024 08:43:12+02:00",
    "Kirjeitse",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon nähtävilläolo 1`] = `
Array [
  Array [
    "Kuulutus suunnitelman nähtäville asettamisesta",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "21.02.2024",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistönomistajien tiedotus Suomi.fi -palvelulla",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi Sukunimi",
    "Osoite 1",
    "00100",
    "Helsinki",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
  Array [
    "123-456-1-291",
    "Etunimi2 Sukunimi2",
    "Osoite 2",
    "01600",
    "Vantaa",
    "21.02.2024 10:01:12+02:00",
    "Suomi.fi",
  ],
]
`;

exports[`tiedotettavatExcel tallenna kiinteistön omistajat excel tiedostoon nähtävilläolo 2`] = `
Array [
  Array [
    "Kuulutus suunnitelman nähtäville asettamisesta",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "21.02.2024",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistönomistajien tiedotus muilla tavoin",
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    "Kiinteistötunnus",
    "Omistajan nimi",
    "Postiosoite",
    "Postinumero",
    "Postitoimipaikka",
    "Tiedot haettu",
    "Tiedotustapa",
  ],
  Array [
    "123-456-1-291",
    "Etunimi3 Sukunimi3",
    "Osoite 3",
    "02600",
    "Espoo",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "123-456-1-291",
    "Yritys Oy Ab",
    "Osoite 4",
    "96101",
    "Rovaniemi",
    "21.02.2024 10:01:12+02:00",
    "Kirjeitse",
  ],
  Array [
    "41-56-1-0",
    null,
    null,
    null,
    null,
    "22.02.2024 08:43:12+02:00",
    "Kirjeitse",
  ],
]
`;
